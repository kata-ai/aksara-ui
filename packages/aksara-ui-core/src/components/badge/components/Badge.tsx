import * as React from 'react';
import styled from 'styled-components';

import BaseBadgeStyles, { BaseBadgeProps } from '../styles';
import { Text, TextProps } from '../../../foundations';

export interface BadgeProps extends BaseBadgeProps, TextProps {
  /** Additional CSS classes to give to the component. */
  className?: string;
  /** Additional CSS properties to give to the component. */
  style?: React.CSSProperties;
}

const Root = styled(Text)<BadgeProps>`
  ${BaseBadgeStyles}
`;

/** Badge text for entity, status, etc. */
const Badge: React.FC<BadgeProps> = ({ className, style, variant, children, ...rest }) => {
  return (
    <Root className={className} style={style} variant={variant} scale={200} {...rest}>
      {children}
    </Root>
  );
};

Badge.defaultProps = {
  variant: 'base',
  className: undefined,
  style: undefined,
};

Badge.displayName = 'Badge';

export default Badge;
