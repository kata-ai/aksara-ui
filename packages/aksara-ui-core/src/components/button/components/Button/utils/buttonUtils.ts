import { css } from 'styled-components';
import { colors } from '../../../../../utils/variables';
import { ButtonBaseProps } from '../types';
import { componentStyles } from '../../../../../utils';

export const ButtonDefault = css`
  color: ${colors.grey08};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonPrimary = css`
  background-color: ${colors.blue06};
  color: ${colors.white};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.blue07};
      color: ${colors.white};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonSupport = css`
  border-color: ${colors.grey04};
  background-color: ${colors.grey01};
  color: ${colors.grey08};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.grey03};
      border-color: ${colors.grey04};
      color: ${colors.blue08};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }

    &.selected {
      background-color: ${colors.blue01};
      color: ${colors.blue08};
      border-color: ${colors.blue04};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey01};
    color: ${colors.grey04};
  }
`;

export const ButtonDestructive = css`
  background-color: ${colors.red07};
  color: ${colors.grey01};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.red08};
      color: ${colors.grey01};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.red09};
      border-color: ${colors.red07};
      color: ${colors.grey01};
      box-shadow: 0 0 2px 1px rgba(186, 19, 9, 0.4), 0 0 0 1px ${colors.red07};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonGhost = css`
  color: ${colors.grey08};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.grey03};
      color: ${colors.blue08};
      border-color: ${colors.blue06};
      box-shadow: 0 0 2px 1px rgba(0, 111, 230, 0.4), 0 0 0 1px ${colors.blue06};
    }
  }

  &:disabled,
  &.disabled {
    border-color: ${colors.grey04};
    background-color: ${colors.grey04};
    color: ${colors.grey01};
  }
`;

export const ButtonSmall = (props: ButtonBaseProps) => css`
  height: 32px;
  padding: 0 12px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 40px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 40px;'}
`;

export const ButtonMedium = (props: ButtonBaseProps) => css`
  height: 40px;
  padding: 0 16px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 44px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 44px;'}
`;

export const ButtonLarge = (props: ButtonBaseProps) => css`
  height: 48px;
  padding: 0 24px;
  font-size: ${componentStyles.text[300].fontSize};
  line-height: ${componentStyles.text[300].lineHeight};
  border-radius: 4px;

  ${props.icon && props.iconPosition === 'left' && 'padding-left: 44px;'}
  ${props.icon && props.iconPosition === 'right' && 'padding-right: 44px;'}
`;

export const InverseButton = css`
  border-color: ${colors.white};
  color: ${colors.white};

  &:disabled,
  &.disabled {
    opacity: 0.5;
  }
`;
