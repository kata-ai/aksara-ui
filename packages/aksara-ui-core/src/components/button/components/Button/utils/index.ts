export { default as renderButtonChildren } from './renderButtonChildren';
export { default as renderButtonIcon } from './renderButtonIcon';
export * from './buttonUtils';
