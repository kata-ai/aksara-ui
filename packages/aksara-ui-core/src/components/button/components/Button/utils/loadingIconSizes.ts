import { ButtonSizes } from '../types';

const loadingIconSizes = (size?: ButtonSizes) => {
  switch (size) {
    case 48:
      return 40;
    case 40:
      return 32;
    case 32:
      return 24;
    default:
      return 32;
  }
};

export default loadingIconSizes;
