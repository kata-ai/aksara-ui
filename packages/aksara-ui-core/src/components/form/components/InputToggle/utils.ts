/* eslint-disable import/prefer-default-export */
import { ToggleSizes, ToggleSizeObject } from './types';

export function toggleButtonSizes(size: ToggleSizes): ToggleSizeObject {
  if (size === 'sm') {
    return {
      width: 32,
      height: 16,
      switch: 12,
    };
  }

  return {
    width: 48,
    height: 24,
    switch: 20,
  };
}
