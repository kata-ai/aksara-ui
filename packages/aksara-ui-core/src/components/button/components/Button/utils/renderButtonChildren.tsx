import * as React from 'react';
import { colors } from '../../../../../utils';
import { InvisibleText, LoaderCircle } from '../components';
import { ButtonBaseProps, ButtonSizes } from '../types';

interface ButtonChildrenProps extends Pick<ButtonBaseProps, 'variant'> {
  isLoading?: boolean;
  size?: ButtonSizes;
  children?: React.ReactNode;
}

const loadingIconSizes = (size?: ButtonSizes) => {
  switch (size) {
    case 48:
      return 40;
    case 40:
      return 32;
    case 32:
      return 24;
    default:
      return 32;
  }
};

const renderButtonChildren = ({ isLoading, size, variant, children }: ButtonChildrenProps) => {
  if (isLoading) {
    return (
      <>
        <LoaderCircle
          size={loadingIconSizes(size)}
          buttonSize={size}
          spinnerColor={variant === 'outline' ? colors.grey04 : colors.white}
        />
        <InvisibleText>{children}</InvisibleText>
      </>
    );
  }

  return children;
};

export default renderButtonChildren;
