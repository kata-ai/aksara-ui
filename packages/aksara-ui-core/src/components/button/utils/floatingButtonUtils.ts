import { css } from 'styled-components';
import { colors } from '../../../utils';
import { ButtonSizes } from '../types';

export function floatingButtonSizes(size?: ButtonSizes) {
  switch (size) {
    case 'lg':
      return 64;
    case 'md':
      return 40;
    case 'sm':
      return 32;
    default:
      return 40;
  }
}

export const FloatingButtonSmall = css`
  height: 32px;
  width: 32px;
  font-size: 13px;
  border-radius: 32px;

  i:before {
    font-size: 13px !important;
  }

  svg {
    width: 20px !important;
    height: 20px !important;
  }
`;

export const FloatingButtonMedium = css`
  height: 40px;
  width: 40px;
  font-size: 16px;
  border-radius: 40px;

  i:before {
    font-size: 16px !important;
  }

  svg {
    width: 24px !important;
    height: 24px !important;
  }
`;

export const FloatingButtonLarge = css`
  height: 64px;
  width: 64px;
  font-size: 24px;
  border-radius: 64px;

  i:before {
    font-size: 24px !important;
  }

  svg {
    width: 32px !important;
    height: 32px !important;
  }
`;

export const FloatingButtonPrimary = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.blue08};
    }
  }
`;

export const FloatingButtonSuccess = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.green07};
    }
  }
`;

export const FloatingButtonWarning = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.yellow07};
    }
  }
`;

export const FloatingButtonDestructive = css`
  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus,
    &:active,
    &.active {
      color: ${colors.red07};
    }
  }
`;
