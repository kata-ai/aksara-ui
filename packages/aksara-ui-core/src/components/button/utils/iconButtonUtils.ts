import { css } from 'styled-components';
import { lighten } from 'polished';
import { themeGet } from '@styled-system/theme-get';
import { colors } from '../../../utils/variables';
import { IconButtonSizes } from '../types';

export function iconButtonSizes(size?: IconButtonSizes) {
  switch (size) {
    case 'lg':
      return 40;
    case 'md':
      return 32;
    case 'sm':
      return 24;
    default:
      return 32;
  }
}

export function iconSizes(size?: IconButtonSizes) {
  switch (size) {
    case 'lg':
      return 24;
    case 'md':
      return 16;
    case 'sm':
      return 12;
    default:
      return 20;
  }
}

export const IconButtonDefault = css`
  background-color: ${colors.gray10};
  color: ${colors.gray50};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.gray20};
      color: ${colors.gray50};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.gray70};
      color: ${colors.white};
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${lighten(0.02, colors.gray10)};
    color: ${colors.gray30};
  }
`;

export const IconButtonPrimary = css`
  background-color: ${colors.cobalt};
  color: ${colors.white};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${colors.semiCobalt};
      color: ${colors.white};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.darkCobalt};
      border-color: ${colors.darkCobalt};
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${lighten(0.02, colors.gray10)};
    color: ${colors.gray30};
  }
`;

export const IconButtonSupport = css`
  background-color: ${themeGet('colors.grey01', colors.grey01)};
  color: ${themeGet('colors.grey08', colors.grey08)};
  border-color: ${themeGet('colors.grey04', colors.grey04)};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: ${themeGet('colors.grey03', colors.grey03)};
      color: ${themeGet('colors.blue08', colors.blue08)};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${themeGet('colors.grey03', colors.grey03)};
      color: ${themeGet('colors.blue08', colors.blue08)};
      border-width: 2px;
      border-color: ${themeGet('colors.blue06', colors.blue06)};
      box-shadow: 0px 0px 2px rgba(0, 111, 230, 0.5);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${themeGet('colors.grey01', colors.grey01)};
    color: ${themeGet('colors.grey04', colors.grey04)};
    border-color: ${themeGet('colors.grey04', colors.grey04)};
  }
`;

export const IconButtonDestructive = css`
  background-color: ${colors.red};
  color: ${colors.white};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover,
    &:focus,
    &.focus {
      background-color: #e84f4c;
      color: ${colors.white};
    }

    &:focus,
    &.focus,
    &:active,
    &.active {
      background-color: ${colors.red};
      box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
    }
  }

  &:disabled,
  &.disabled {
    background-color: ${lighten(0.02, colors.gray10)};
    color: ${colors.gray30};
  }
`;

export const IconButtonGhost = css`
  color: ${themeGet('colors.grey08', colors.grey08)};

  &:not(:disabled):not(.disabled) {
    &:hover,
    &.hover {
      color: ${themeGet('colors.blue08', colors.blue08)};
    }

    &:focus,
    &.focus {
      color: ${themeGet('colors.blue08', colors.blue08)};
      border-color: ${themeGet('colors.blue04', colors.blue04)};
      box-shadow: 0px 0px 2px rgba(0, 111, 230, 0.5);
    }

    &:active,
    &.active {
      border-color: ${themeGet('colors.indigo02', colors.indigo02)};
    }
  }

  &:disabled,
  &.disabled {
    color: ${colors.grey04};
  }
`;
