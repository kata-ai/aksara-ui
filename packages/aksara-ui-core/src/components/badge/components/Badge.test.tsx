import * as React from 'react';
import { render } from '@testing-library/react';

import { colors } from '../../../utils';
import Badge from './Badge';

describe('compoents/Badge', () => {
  describe('<Badge />', () => {
    test('renders with children', () => {
      const { container } = render(<Badge>test badge</Badge>);

      expect(container).toHaveTextContent('test badge');
    });

    test('renders with additional variants', () => {
      const { container } = render(<Badge variant="green">test badge</Badge>);

      expect(container.firstChild).toHaveStyleRule('background-color', colors.green03);
      expect(container.firstChild).toHaveStyleRule('color', colors.grey08);
    });
  });
});
