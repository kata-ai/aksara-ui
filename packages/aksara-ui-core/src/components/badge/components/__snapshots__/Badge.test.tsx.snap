// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compoents/Badge <Badge /> renders with additional variants 1`] = `
.c0 {
  display: inline-block;
  padding: 8px 12px;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border: 1px solid transparent;
  border-radius: 4px;
  background-color: #c1e0b7;
  color: #313f4e;
}

<span
  class="sc-AxiKw c0"
  scale="200"
>
  test badge
</span>
`;
